File: import_leads.py
#!/usr/bin/env python3
"""
Script to import lead data from Excel files into the CRM database.
"""
import mysql.connector
import os
from datetime import datetime
import json

# Database connection
db_url = os.environ.get('DATABASE_URL', '')
# Parse DATABASE_URL: mysql://user:pass@host:port/database
if db_url.startswith('mysql://'):
    db_url = db_url[8:]
    user_pass, host_db = db_url.split('@')
    user, password = user_pass.split(':')
    host_port, database = host_db.split('/')
    if ':' in host_port:
        host, port = host_port.split(':')
        port = int(port)
    else:
        host = host_port
        port = 3306
else:
    # Fallback to individual env vars
    host = os.environ.get('DB_HOST', 'localhost')
    port = int(os.environ.get('DB_PORT', 3306))
    user = os.environ.get('DB_USER', 'root')
    password = os.environ.get('DB_PASSWORD', '')
    database = os.environ.get('DB_NAME', 'crm')

print(f"Connecting to database at {host}:{port}/{database}")

conn = mysql.connector.connect(
    host=host,
    port=port,
    user=user,
    password=password,
    database=database,
    ssl_disabled=False
)
cursor = conn.cursor(dictionary=True)

# Lead 1: Margate Property
margate_lead = {
    'address': '5771 NW 24th Ct',
    'city': 'Margate',
    'state': 'FL',
    'zip': '33063',
    'propertyType': 'Single Family Home',
    'yearBuilt': 1959,
    'bedrooms': 3,
    'bathrooms': 1,
    'squareFeet': 920,
    'lotSize': '7,218 sq. ft.',
    'ownerName': 'Florence M. Giroux',
    'ownerPhone': '(561) 699-2623',
    'ownerPhone2': '(954) 972-1885',
    'ownerEmail': 'f******x@yahoo.com',
    'ownerAddress': '5771 NW 24th Ct, Margate, FL 33063',
    'occupancy': 'Owner-Occupied',
    'estimatedValue': 335530,
    'taxAmount': 5489,
    'taxStatus': 'Current and Paid',
    'codeViolations': False,
    'probateStatus': 'No Probate Case Found',
    'leadTemperature': 'warm',
    'status': 'new_prospect',
    'notes': '''KEY FINDINGS:
1. Florence M. Giroux (age 84) is current owner and active resident
2. No probate case or deceased owner identified
3. Property is owner-occupied and well-maintained
4. Current market value estimated at $335,530
5. Property taxes current and paid ($5,489 in 2024)
6. No code violations or enforcement actions found
7. Property is off-market and not currently for sale
8. Secondary owner listed (possibly spouse)

RECOMMENDED NEXT STEPS:
1. Contact Florence M. Giroux at (561) 699-2623 or (954) 972-1885
2. Obtain certified deed from Broward County Official Records
3. Search for mortgage and lien information in official records'''
}

# Lead 2: Miramar Property
miramar_lead = {
    'address': '6717 Arbor Dr',
    'city': 'Miramar',
    'state': 'FL',
    'zip': '33023',
    'propertyType': 'Single Family Home',
    'yearBuilt': 1965,
    'bedrooms': 3,
    'bathrooms': 2,
    'squareFeet': 1544,
    'lotSize': '6,505 sq. ft.',
    'ownerName': 'Len D. Sumlar',
    'ownerPhone': '',
    'ownerAddress': 'Miami-Dade County (Employer)',
    'occupancy': 'Tenant-Occupied',
    'estimatedValue': 459014,
    'taxAmount': 4750,
    'taxStatus': 'Current and Paid',
    'codeViolations': False,
    'probateStatus': 'No Probate Case Found',
    'leadTemperature': 'warm',
    'status': 'new_prospect',
    'notes': '''KEY FINDINGS:
1. Property is currently owned by Len D. Sumlar (purchased 2019 for $200,000)
2. No probate case has been identified for this property
3. Current market value estimated at $459,014 - $544,857 (significant appreciation since 1994)
4. Property is off-market and not currently for sale
5. Current residents include Davion Hudson (age 36) and family members
6. Property has no code violations or enforcement actions
7. Property taxes are current and paid
8. Last recorded sale was February 2, 1994 for $85,000
9. Property appreciation of 440-540% since 1994 sale

RECOMMENDED NEXT STEPS:
1. Contact Len D. Sumlar through Miami-Dade County Medical Examiner's office
2. Contact Davion Hudson at (954) 966-0140 to verify resident status
3. Conduct title search to verify clear ownership'''
}

# Miramar contacts
miramar_contacts = [
    {'name': 'Len D. Sumlar', 'relationship': 'Current Owner', 'phone': '', 'notes': 'Manager, Morgue Bureau Operations, Medical Examiner'},
    {'name': 'Davion Hudson', 'relationship': 'Current Resident', 'phone': '(954) 966-0140', 'notes': 'Democratic Party voter; born November 1989'},
    {'name': 'Frona S. Hudson', 'relationship': 'Resident/Family', 'phone': '(954) 986-8378', 'notes': 'Born November 1961; family member'},
    {'name': 'Donovan Hudson', 'relationship': 'Resident/Family', 'phone': '(954) 986-8378', 'notes': 'Family member; living at property'},
]

def insert_property(lead):
    """Insert or update a property in the database."""
    # Check if property already exists
    cursor.execute(
        "SELECT id FROM properties WHERE address = %s AND city = %s",
        (lead['address'], lead['city'])
    )
    existing = cursor.fetchone()
    
    now = datetime.now()
    
    if existing:
        property_id = existing['id']
        print(f"Property already exists with ID {property_id}, updating...")
        cursor.execute("""
            UPDATE properties SET
                state = %s,
                zip = %s,
                property_type = %s,
                year_built = %s,
                bedrooms = %s,
                bathrooms = %s,
                square_feet = %s,
                owner_name = %s,
                estimated_value = %s,
                tax_amount = %s,
                lead_temperature = %s,
                status = %s,
                updated_at = %s
            WHERE id = %s
        """, (
            lead['state'],
            lead['zip'],
            lead['propertyType'],
            lead['yearBuilt'],
            lead['bedrooms'],
            lead['bathrooms'],
            lead['squareFeet'],
            lead['ownerName'],
            lead['estimatedValue'],
            lead['taxAmount'],
            lead['leadTemperature'],
            lead['status'],
            now,
            property_id
        ))
    else:
        print(f"Creating new property: {lead['address']}, {lead['city']}")
        cursor.execute("""
            INSERT INTO properties (
                address, city, state, zip, property_type, year_built,
                bedrooms, bathrooms, square_feet, owner_name,
                estimated_value, tax_amount, lead_temperature, status,
                created_at, updated_at
            ) VALUES (%s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s)
        """, (
            lead['address'],
            lead['city'],
            lead['state'],
            lead['zip'],
            lead['propertyType'],
            lead['yearBuilt'],
            lead['bedrooms'],
            lead['bathrooms'],
            lead['squareFeet'],
            lead['ownerName'],
            lead['estimatedValue'],
            lead['taxAmount'],
            lead['leadTemperature'],
            lead['status'],
            now,
            now
        ))
        property_id = cursor.lastrowid
    
    return property_id

def insert_deep_search(property_id, lead):
    """Insert or update deep search data for a property."""
    cursor.execute(
        "SELECT id FROM property_deep_search WHERE property_id = %s",
        (property_id,)
    )
    existing = cursor.fetchone()
    
    now = datetime.now()
    
    # Prepare deep search data
    occupancy = lead.get('occupancy', '')
    has_code_violation = lead.get('codeViolations', False)
    zillow_estimate = lead.get('estimatedValue', 0)
    research_notes = lead.get('notes', '')
    
    if existing:
        print(f"Updating deep search for property {property_id}")
        cursor.execute("""
            UPDATE property_deep_search SET
                occupancy = %s,
                has_code_violation = %s,
                zillow_estimate = %s,
                research_notes = %s,
                updated_at = %s
            WHERE property_id = %s
        """, (
            occupancy,
            has_code_violation,
            zillow_estimate,
            research_notes,
            now,
            property_id
        ))
    else:
        print(f"Creating deep search for property {property_id}")
        cursor.execute("""
            INSERT INTO property_deep_search (
                property_id, occupancy, has_code_violation,
                zillow_estimate, research_notes, created_at, updated_at
            ) VALUES (%s, %s, %s, %s, %s, %s, %s)
        """, (
            property_id,
            occupancy,
            has_code_violation,
            zillow_estimate,
            research_notes,
            now,
            now
        ))

def insert_contact(property_id, contact):
    """Insert a contact for a property."""
    # Check if contact already exists
    cursor.execute(
        "SELECT id FROM property_contacts WHERE property_id = %s AND name = %s",
        (property_id, contact['name'])
    )
    existing = cursor.fetchone()
    
    now = datetime.now()
    
    if existing:
        print(f"Contact {contact['name']} already exists, updating...")
        cursor.execute("""
            UPDATE property_contacts SET
                relationship = %s,
                phone1 = %s,
                notes = %s,
                updated_at = %s
            WHERE id = %s
        """, (
            contact.get('relationship', ''),
            contact.get('phone', ''),
            contact.get('notes', ''),
            now,
            existing['id']
        ))
    else:
        print(f"Creating contact: {contact['name']}")
        cursor.execute("""
            INSERT INTO property_contacts (
                property_id, name, relationship, phone1, notes,
                created_at, updated_at
            ) VALUES (%s, %s, %s, %s, %s, %s, %s)
        """, (
            property_id,
            contact['name'],
            contact.get('relationship', ''),
            contact.get('phone', ''),
            contact.get('notes', ''),
            now,
            now
        ))

# Import Margate lead
print("\n=== Importing Margate Lead ===")
margate_id = insert_property(margate_lead)
insert_deep_search(margate_id, margate_lead)
# Add owner as contact
insert_contact(margate_id, {
    'name': margate_lead['ownerName'],
    'relationship': 'Owner',
    'phone': margate_lead['ownerPhone'],
    'notes': f"Age 84. Secondary phone: {margate_lead['ownerPhone2']}. Email: {margate_lead['ownerEmail']}"
})
print(f"Margate lead imported with ID: {margate_id}")

# Import Miramar lead
print("\n=== Importing Miramar Lead ===")
miramar_id = insert_property(miramar_lead)
insert_deep_search(miramar_id, miramar_lead)
# Add all contacts
for contact in miramar_contacts:
    insert_contact(miramar_id, contact)
print(f"Miramar lead imported with ID: {miramar_id}")

conn.commit()
cursor.close()
conn.close()

print("\n=== Import Complete ===")
print(f"Margate Property ID: {margate_id}")
print(f"Miramar Property ID: {miramar_id}")

File: import_leads.mjs
#!/usr/bin/env node
/**
 * Script to import lead data from Excel files into the CRM database.
 * Run with: node import_leads.mjs
 */
import mysql from 'mysql2/promise';
import dotenv from 'dotenv';

dotenv.config();

const DATABASE_URL = process.env.DATABASE_URL;

// Parse DATABASE_URL
function parseDbUrl(url) {
  // mysql://user:pass@host:port/database
  const match = url.match(/mysql:\/\/([^:]+):([^@]+)@([^:]+):(\d+)\/(.+)/);
  if (!match) throw new Error('Invalid DATABASE_URL format');
  return {
    user: match[1],
    password: match[2],
    host: match[3],
    port: parseInt(match[4]),
    database: match[5].split('?')[0]
  };
}

const dbConfig = parseDbUrl(DATABASE_URL);
console.log(`Connecting to database at ${dbConfig.host}:${dbConfig.port}/${dbConfig.database}`);

// Lead 1: Margate Property
const margateProperty = {
  addressLine1: '5771 NW 24th Ct',
  city: 'Margate',
  state: 'FL',
  zipcode: '33063',
  propertyType: 'Single Family Home',
  yearBuilt: 1959,
  totalBedrooms: 3,
  totalBaths: 1,
  buildingSquareFeet: 920,
  owner1Name: 'Florence M. Giroux',
  estimatedValue: 335530,
  taxAmount: 5489,
  leadTemperature: 'WARM',
  trackingStatus: 'Not Visited',
  ownerVerified: 1,
  ownerLocation: 'Same as Property'
};

const margateDeepSearch = {
  occupancy: 'Owner-Occupied',
  mlsStatus: 'Off Market',
  zillowEstimate: 335530,
  hasCodeViolation: 0,
  hasLiens: 0,
  recordsCheckedNotes: `KEY FINDINGS:
1. Florence M. Giroux (age 84) is current owner and active resident
2. No probate case or deceased owner identified
3. Property is owner-occupied and well-maintained
4. Current market value estimated at $335,530
5. Property taxes current and paid ($5,489 in 2024)
6. No code violations or enforcement actions found
7. Property is off-market and not currently for sale
8. Secondary owner listed (possibly spouse)

RECOMMENDED NEXT STEPS:
1. Contact Florence M. Giroux at (561) 699-2623 or (954) 972-1885
2. Obtain certified deed from Broward County Official Records
3. Search for mortgage and lien information in official records

SOURCES SEARCHED:
- Broward County Property Appraiser Database
- Broward County Clerk of Courts (Probate Case Search)
- Broward County Official Records (Deeds and Liens)
- City of Margate Code Enforcement Database
- Redfin, Zillow, Realtor.com (Property Valuation)`,
  recordsChecked: JSON.stringify(["County", "Property Taxes", "Code Enforcement", "Probate Court"]),
  probateFinds: JSON.stringify(["No Probate Case Found"]),
  probateNotes: 'Property Not in Probate - No deceased owner identified'
};

const margateContacts = [
  {
    name: 'Florence M. Giroux',
    relationship: 'Owner',
    age: 84,
    phone1: '(561) 699-2623',
    phone1Type: 'Mobile',
    phone2: '(954) 972-1885',
    phone2Type: 'Landline',
    email1: 'f******x@yahoo.com',
    currentAddress: '5771 NW 24th Ct, Margate, FL 33063',
    isDecisionMaker: 1
  },
  {
    name: 'M****** G*****',
    relationship: 'Secondary Owner (Possibly Spouse)',
    currentAddress: '5771 NW 24th Ct, Margate, FL 33063'
  }
];

// Lead 2: Miramar Property
const miramarProperty = {
  addressLine1: '6717 Arbor Dr',
  city: 'Miramar',
  state: 'FL',
  zipcode: '33023',
  propertyType: 'Single Family Home',
  yearBuilt: 1965,
  totalBedrooms: 3,
  totalBaths: 2,
  buildingSquareFeet: 1544,
  owner1Name: 'Len D. Sumlar',
  estimatedValue: 459014,
  taxAmount: 4750,
  leadTemperature: 'WARM',
  trackingStatus: 'Not Visited',
  ownerVerified: 0,
  salePrice: 200000,
  ownerLocation: 'Miami-Dade County'
};

const miramarDeepSearch = {
  occupancy: 'Tenant-Occupied',
  mlsStatus: 'Off Market',
  zillowEstimate: 459014,
  dealMachineEstimate: 544857,
  ourEstimate: 501936,
  hasCodeViolation: 0,
  hasLiens: 0,
  recordsCheckedNotes: `KEY FINDINGS:
1. Property is currently owned by Len D. Sumlar (purchased 2019 for $200,000)
2. No probate case has been identified for this property
3. Current market value estimated at $459,014 - $544,857 (significant appreciation since 1994)
4. Property is off-market and not currently for sale
5. Current residents include Davion Hudson (age 36) and family members
6. Property has no code violations or enforcement actions
7. Property taxes are current and paid
8. Last recorded sale was February 2, 1994 for $85,000
9. Property appreciation of 440-540% since 1994 sale

RECOMMENDED NEXT STEPS:
1. Contact Len D. Sumlar through Miami-Dade County Medical Examiner's office
2. Contact Davion Hudson at (954) 966-0140 to verify resident status
3. Conduct title search to verify clear ownership

SOURCES SEARCHED:
- Broward County Property Appraiser Database
- Broward County Clerk of Courts (Probate Case Search)
- Broward County Official Records (Deeds and Liens)
- City of Miramar Code Enforcement Database
- Redfin, Zillow, Realtor.com, Homes.com (Property Valuation)
- FastPeopleSearch, Voter Records (Resident Information)`,
  recordsChecked: JSON.stringify(["County", "Property Taxes", "Code Enforcement", "Probate Court", "Voter Records"]),
  probateFinds: JSON.stringify(["No Probate Case Found"]),
  probateNotes: 'No probate case identified in Broward County system',
  deedType: JSON.stringify([
    { type: "Warranty Deed", deedDate: "2019-01-01", amount: 200000, notes: "Len D. Sumlar purchase" },
    { type: "Previous Sale", deedDate: "1994-02-02", amount: 85000, notes: "Last recorded sale" }
  ])
};

const miramarContacts = [
  {
    name: 'Len D. Sumlar',
    relationship: 'Owner',
    currentAddress: 'Miami-Dade County (Employer - Medical Examiner)',
    isDecisionMaker: 1,
    flags: 'Manager, Morgue Bureau Operations, Medical Examiner'
  },
  {
    name: 'Davion Hudson',
    relationship: 'Current Resident',
    age: 36,
    phone1: '(954) 966-0140',
    phone1Type: 'Mobile',
    currentAddress: '6717 Arbor Dr, Miramar, FL 33023',
    flags: 'Democratic Party voter; born November 1989'
  },
  {
    name: 'Frona S. Hudson',
    relationship: 'Resident/Family',
    age: 63,
    phone1: '(954) 986-8378',
    phone1Type: 'Landline',
    currentAddress: '6717 Arbor Dr, Miramar, FL 33023',
    flags: 'Born November 1961; family member'
  },
  {
    name: 'Donovan Hudson',
    relationship: 'Resident/Family',
    age: 36,
    phone1: '(954) 986-8378',
    phone1Type: 'Landline',
    phone2: '(954) 966-0140',
    phone2Type: 'Mobile',
    currentAddress: '6717 Arbor Dr, Miramar, FL 33023',
    flags: 'Family member; living at property'
  }
];

async function main() {
  const conn = await mysql.createConnection({
    host: dbConfig.host,
    port: dbConfig.port,
    user: dbConfig.user,
    password: dbConfig.password,
    database: dbConfig.database,
    ssl: { rejectUnauthorized: false }
  });

  console.log('Connected to database');

  async function insertProperty(prop) {
    // Check if property exists
    const [existing] = await conn.execute(
      'SELECT id FROM properties WHERE addressLine1 = ? AND city = ?',
      [prop.addressLine1, prop.city]
    );

    if (existing.length > 0) {
      const id = existing[0].id;
      console.log(`Property ${prop.addressLine1} already exists with ID ${id}, updating...`);
      await conn.execute(`
        UPDATE properties SET
          state = ?, zipcode = ?, propertyType = ?, yearBuilt = ?,
          totalBedrooms = ?, totalBaths = ?, buildingSquareFeet = ?,
          owner1Name = ?, estimatedValue = ?, taxAmount = ?,
          leadTemperature = ?, trackingStatus = ?, ownerVerified = ?,
          ownerLocation = ?, salePrice = ?, updatedAt = NOW()
        WHERE id = ?
      `, [
        prop.state, prop.zipcode, prop.propertyType, prop.yearBuilt,
        prop.totalBedrooms, prop.totalBaths, prop.buildingSquareFeet,
        prop.owner1Name, prop.estimatedValue, prop.taxAmount,
        prop.leadTemperature, prop.trackingStatus, prop.ownerVerified,
        prop.ownerLocation || null, prop.salePrice || null, id
      ]);
      return id;
    } else {
      console.log(`Creating new property: ${prop.addressLine1}, ${prop.city}`);
      const [result] = await conn.execute(`
        INSERT INTO properties (
          addressLine1, city, state, zipcode, propertyType, yearBuilt,
          totalBedrooms, totalBaths, buildingSquareFeet, owner1Name,
          estimatedValue, taxAmount, leadTemperature, trackingStatus,
          ownerVerified, ownerLocation, salePrice, createdAt, updatedAt
        ) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, NOW(), NOW())
      `, [
        prop.addressLine1, prop.city, prop.state, prop.zipcode, prop.propertyType,
        prop.yearBuilt, prop.totalBedrooms, prop.totalBaths, prop.buildingSquareFeet,
        prop.owner1Name, prop.estimatedValue, prop.taxAmount, prop.leadTemperature,
        prop.trackingStatus, prop.ownerVerified, prop.ownerLocation || null, prop.salePrice || null
      ]);
      return result.insertId;
    }
  }

  async function insertDeepSearch(propertyId, ds) {
    const [existing] = await conn.execute(
      'SELECT id FROM propertyDeepSearch WHERE propertyId = ?',
      [propertyId]
    );

    if (existing.length > 0) {
      console.log(`Updating deep search for property ${propertyId}`);
      await conn.execute(`
        UPDATE propertyDeepSearch SET
          occupancy = ?, mlsStatus = ?, zillowEstimate = ?,
          dealMachineEstimate = ?, ourEstimate = ?,
          hasCodeViolation = ?, hasLiens = ?,
          recordsCheckedNotes = ?, recordsChecked = ?,
          probateFinds = ?, probateNotes = ?, deedType = ?,
          updatedAt = NOW()
        WHERE propertyId = ?
      `, [
        ds.occupancy, ds.mlsStatus, ds.zillowEstimate,
        ds.dealMachineEstimate || null, ds.ourEstimate || null,
        ds.hasCodeViolation, ds.hasLiens,
        ds.recordsCheckedNotes, ds.recordsChecked || null,
        ds.probateFinds || null, ds.probateNotes || null, ds.deedType || null,
        propertyId
      ]);
    } else {
      console.log(`Creating deep search for property ${propertyId}`);
      await conn.execute(`
        INSERT INTO propertyDeepSearch (
          propertyId, occupancy, mlsStatus, zillowEstimate,
          dealMachineEstimate, ourEstimate,
          hasCodeViolation, hasLiens,
          recordsCheckedNotes, recordsChecked,
          probateFinds, probateNotes, deedType,
          createdAt, updatedAt
        ) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, NOW(), NOW())
      `, [
        propertyId, ds.occupancy, ds.mlsStatus, ds.zillowEstimate,
        ds.dealMachineEstimate || null, ds.ourEstimate || null,
        ds.hasCodeViolation, ds.hasLiens,
        ds.recordsCheckedNotes, ds.recordsChecked || null,
        ds.probateFinds || null, ds.probateNotes || null, ds.deedType || null
      ]);
    }
  }

  async function insertContact(propertyId, contact) {
    const [existing] = await conn.execute(
      'SELECT id FROM contacts WHERE propertyId = ? AND name = ?',
      [propertyId, contact.name]
    );

    if (existing.length > 0) {
      console.log(`Contact ${contact.name} already exists, updating...`);
      await conn.execute(`
        UPDATE contacts SET
          relationship = ?, age = ?, phone1 = ?, phone1Type = ?,
          phone2 = ?, phone2Type = ?, email1 = ?,
          currentAddress = ?, isDecisionMaker = ?, flags = ?,
          updatedAt = NOW()
        WHERE id = ?
      `, [
        contact.relationship || null, contact.age || null,
        contact.phone1 || null, contact.phone1Type || null,
        contact.phone2 || null, contact.phone2Type || null,
        contact.email1 || null, contact.currentAddress || null,
        contact.isDecisionMaker || 0, contact.flags || null,
        existing[0].id
      ]);
    } else {
      console.log(`Creating contact: ${contact.name}`);
      await conn.execute(`
        INSERT INTO contacts (
          propertyId, name, relationship, age, phone1, phone1Type,
          phone2, phone2Type, email1, currentAddress, isDecisionMaker, flags,
          createdAt, updatedAt
        ) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, NOW(), NOW())
      `, [
        propertyId, contact.name, contact.relationship || null, contact.age || null,
        contact.phone1 || null, contact.phone1Type || null,
        contact.phone2 || null, contact.phone2Type || null,
        contact.email1 || null, contact.currentAddress || null,
        contact.isDecisionMaker || 0, contact.flags || null
      ]);
    }
  }

  try {
    // Import Margate lead
    console.log('\n=== Importing Margate Lead ===');
    const margateId = await insertProperty(margateProperty);
    await insertDeepSearch(margateId, margateDeepSearch);
    for (const contact of margateContacts) {
      await insertContact(margateId, contact);
    }
    console.log(`Margate lead imported with ID: ${margateId}`);

    // Import Miramar lead
    console.log('\n=== Importing Miramar Lead ===');
    const miramarId = await insertProperty(miramarProperty);
    await insertDeepSearch(miramarId, miramarDeepSearch);
    for (const contact of miramarContacts) {
      await insertContact(miramarId, contact);
    }
    console.log(`Miramar lead imported with ID: ${miramarId}`);

    console.log('\n=== Import Complete ===');
    console.log(`Margate Property ID: ${margateId}`);
    console.log(`Miramar Property ID: ${miramarId}`);

  } catch (error) {
    console.error('Error importing leads:', error);
  } finally {
    await conn.end();
  }
}

main();
